### ISSUE TITLE
Node-based achecker utility does not produce HTML output

### ISSUE THREAD

**Issue by jamalmazrui** [IBMa/equal-access#241]
After installing accessibility-checker with npm, I tried running achecker with "--outputFormat html" as mentioned in the documentation.  An error results, apparently related to code for creating an output directory.  This happens even if I have first created a "results" subdirectory and passed "--outputFolder results" as a parameter.

I am most interested in HTML output, but JSON did not work either.  On the other hand, CSV output did work.  This is on Windows.

**Comment by tombrunet** [comment]
```
$ achecker --outputFormat html https://altoromutual.com
Failed: https://altoromutual.com

Failing scan details:

Scan: https://altoromutual.com
- Message: Table has no headers identified
  Level: violation
  XPath: /html[1]/body[1]/div[1]/form[1]/table[1]
  Snippet: <table cellspacing="0" cellpadding="0" border="0" width="100%">
  Help: https://able.ibm.com/rules/tools/help/RPT_Table_DataHeadingsAria

...snipped...

$ ls results/https\:/altoromutual.com.html
results/https:/altoromutual.com.html
```

It's working fine for me. Can you give more specifics on how you are running it?

**Comment by jamalmazrui** [comment]
Thanks for your reply.  At a Windows 10 command prompt, I just entered 
the following command:

achecker --outputFormat html https://altoromutual.com

Here is the output:

Error: ENOENT: no such file or directory, mkdir 'results\https:'
     at Object.mkdirSync (fs.js:921:3)
     at writeObjectToFileAsHTML (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\reporters\ACReporterHTML.js:159:16)
     at savePageResults (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\reporters\ACReporterHTML.js:118:9)
     at ACReporterHTML.report (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\reporters\ACReporterHTML.js:54:9)
     at Object.aChecker.sendResultsToReporter (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\ACHelper.js:764:26)
     at Object.aChecker.getComplianceHelperPuppeteer (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\ACHelper.js:626:26)
     at processTicksAndRejections (internal/process/task_queues.js:97:5)
     at async Object.aChecker.getComplianceHelper (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\ACHelper.js:473:20)
     at async C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\bin\achecker.js:195:22 
{
   errno: -4058,
   syscall: 'mkdir',
   code: 'ENOENT',
   path: 'results\\https:'
}
(node:2372) UnhandledPromiseRejectionWarning: Error: ENOENT: no such 
file or directory, mkdir 'results\https:'
     at Object.mkdirSync (fs.js:921:3)
     at writeObjectToFileAsHTML (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\reporters\ACReporterHTML.js:159:16)
     at savePageResults (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\reporters\ACReporterHTML.js:118:9)
     at ACReporterHTML.report (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\reporters\ACReporterHTML.js:54:9)
     at Object.aChecker.sendResultsToReporter (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\ACHelper.js:764:26)
     at Object.aChecker.getComplianceHelperPuppeteer (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\ACHelper.js:626:26)
     at processTicksAndRejections (internal/process/task_queues.js:97:5)
     at async Object.aChecker.getComplianceHelper (C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\lib\ACHelper.js:473:20)
     at async C:\Users\Jamal 
Mazrui\AppData\Roaming\npm\node_modules\accessibility-checker\bin\achecker.js:195:22
(node:2372) UnhandledPromiseRejectionWarning: Unhandled promise 
rejection. This error originated either by throwing inside of an async 
function without a catch block, or by rejecting a promise which was not 
handled with .catch(). To terminate the node process on unhandled 
promise rejection, use the CLI flag `--unhandled-rejections=strict` (see 
https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). 
(rejection id: 2)
(node:2372) [DEP0018] DeprecationWarning: Unhandled promise rejections 
are deprecated. In the future, promise rejections that are not handled 
will terminate the Node.js process with a non-zero exit code.

**Comment by tombrunet** [comment]
I can't seem to find the issue to reference at the moment, but another user had a similar problem because they were running Node 8. Make sure you're using at least Node 10.12. Node did not support recursive directory creation before that version. The latest LTS version is 12.18.4.

**Comment by jamalmazrui** [comment]
I think I have a current version.  I just entered the following command


node -v


Here is the output:


v12.18.4


Can you test on Windows?

**Comment by tombrunet** [comment]
Okay - this is due to Windows not allowing `:` in file / directory names. We'll have to sanitize that label for Windows.

**Comment by tombrunet** [comment]
@jamalmazrui We deployed a new version yesterday, 3.1.0. Please try to install that version and try again.

**Comment by jamalmazrui** [comment]
Thank you for the quick fix.  I found that the achecker command on 
Windows can now create an HTML report.


FYI, I also found inconsistencies in the subdirectory structure and 
naming conventions used when the output format was html vs json vs csv 
-- beyond differences in extensions of output files. For example, the 
html output was a file inside a subdirectory named after the web domain 
tested, with a file name of just ".html" -- unusual for the name of a 
file on Windows and for a file that is not meant to be hidden by 
default.  The json output, on the other hand seemed to be in a file 
named "results.json" (a good name, IMHO) at a higher directory level.


Another FYI is that the documentation for achecker mentions html as an 
output format, besides others, whereas the command-line help only 
mentions json and csv.

**Comment by tombrunet** [comment]
Closing. Will track path inconsistencies in #319
